gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/token/transfer.ts

```diff
@@ -20,28 +21,63 @@ export default class TransferToken extends SolanaCommand {
     this.require(!!args[0], 'Provide a token address')
   }
 
-  execute = async () => {
+  isValidTokenAccount = async (token: Token, account: PublicKey) => {
+    try {
+      const info = await token.getAccountInfo(new PublicKey(account))
+      return !!info.address
+    } catch (e) {
+      return false
+    }
+  }
+
+  makeRawTransaction = async (signer: PublicKey) => {
     const address = this.args[0]
-    const token = new Token(this.provider.connection, new PublicKey(address), TOKEN_PROGRAM_ID, this.wallet.payer)
+
+    const token = new Token(this.provider.connection, new PublicKey(address), TOKEN_PROGRAM_ID, {
+      publicKey: signer,
+      secretKey: Buffer.from([]),
+    })
 
     const from = await Token.getAssociatedTokenAddress(
       ASSOCIATED_TOKEN_PROGRAM_ID,
       TOKEN_PROGRAM_ID,
       token.publicKey,
-      this.wallet.publicKey,
+      signer,
+      true,
+    )
+
+    const destination = new PublicKey(this.flags.to)
+    const amount = new BN(this.flags.amount).mul(new BN(10).pow(new BN(TOKEN_DECIMALS)))
+    this.require(
+      await this.isValidTokenAccount(token, destination),
+      `Destination ${destination.toString()} is not a valid token account`,
     )
 
-    const destination = (await token.getOrCreateAssociatedAccountInfo(new PublicKey(this.flags.to))).address
-    const amount = new BN(this.flags.amount).toNumber()
+    logger.info(
+      `Preparing instruction to send ${amount.toString()} (${this.flags.amount}) Tokens to ${destination.toString()}`,
+    )
+    const ix = Token.createTransferInstruction(TOKEN_PROGRAM_ID, from, destination, signer, [], amount.toNumber())
```

Comment:
`flags.amount` is expected to be a decimal number?
We could potentially lose precision with `amount.toNumber()`, would `amount.toString()` work?

Reply:
The transfer function only supports a `u64` amount apparently